Update docs

mslw opened the pull request:
This is a rewrite of the documentation, which shifts the "dataladification" process from admin docs to user docs. This reflects a change in process compared to initial plans: while DataLad datasets can be created by ICF and placed alongside tarballs, this step will most likely be done by users on their institute local infrastructure.

Consequently, the user docs chapter is now organised into the following sections, and assumes only DICOMs are available from the ICF:

- Browser based access
- DataLad requirements
- Manage DataLad credentials
- Generate DataLad datasets
- Access data with DataLad
- DataLad datasets: advanced features

A new chapter, Reference, is added, and groups two sections:

- Containerized execution
- Script listing

Administrator docs chapter is shortened to a minimum, and describes dataladification steps as optional, linking to the reference parts above.

Majority of the old text is retained with minor changes. Apart from restructuring, the biggest change is the addition of "Generate DataLad datasets" from the point of view of a local user.

mslw commented:
Note: please keep the PR open, aiming to finish it today.

mslw commented:
It would be an awesome test of the updated docs if someone with access credentials to the test data, but a hazy memory of details, builds the docs and tries to generate and then clone a DataLad dataset from DICOMs.

adswa commented on docs/source/user/datalad-generate.rst:
```diff
@@ -0,0 +1,131 @@
+   export STORE_DIR=$PWD/local_dicom_store
+
+
+Deposit visit metadata alongside tarball
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Information required to create a DataLad dataset needs to be extracted
+from the tarball:
+
```
Following through the docs sequentially, I don't think I've come across this singularity image before. I think it would make sense to link to its download page here.

mslw replied:
The 3rd paragraph of this page says:

> The workflow described below uses DataLad with DataLad-Next extension for initial DICOM download and the INM-ICF tools packaged as a Singularity container for subsequent steps (see DataLad requirements).

where "DataLad requirements" is a link to a page that describes things in greater details (and is actually positioned earlier in the User Guide), and links to containerized execution page.

However, your comment makes it apparent that I didn't do a good enough job when trying to compartmentalize the docs (to avoid repetition), and I will add a sentence of two to make up for it.

<hr>

By the way, this points to a small design issue with the tooling. Initially, the Singularity image was just for ICF. ICF would only use DataLad through the scripts in this image. Users would not need the Singularity image, they would clone datasets from ICF using DataLad.

Now, users who want to dataladify datasets using the Singularity image still need to download the tarballs somehow. I decided to suggest `datalad download` for the task, because it interacts with DataLad credentials, that would also be needed for any subsequent dataset content retrieval from ICF. Alternatively, we could recommend `curl -u` followed by Singularity (no need to install DataLad), or `datalad download` followed by running scripts from this repo (no need for Singularity). The former seems unsatisfactory, because any further dataset interaction would need to happen through DataLad anyway. The latter seems unsatisfactory because the Singularity image was introduced to make the ICF tooling independent of changes in DataLad.

adswa commented on docs/source/user/datalad-generate.rst:
```diff
@@ -0,0 +1,131 @@
+
+.. code-block:: bash
+
+   datalad download "https://data.inm-icf.de/<project-ID>/<visit-ID>_dicom.tar <project-ID>/<visit-ID>_dicom.tar"
```
I believe it should go inside the local store?

```suggestion
   datalad download "https://data.inm-icf.de/<project-ID>/<visit-ID>_dicom.tar  local_dicom_store/<project-ID>/<visit-ID>_dicom.tar"
```

mslw replied:
Lol, the double space in the argument makes it download to ` local_dicom_store` instead of `local_dicom_store`.

adswa replied:
> Lol, the double space in the argument makes it download to  local_dicom_store instead of local_dicom_store.

oh no.... :o

adswa commented on docs/source/user/datalad-generate.rst:
```diff
@@ -0,0 +1,131 @@
+Deposit dataset alongside tarball
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+DataLad dataset is created based on the metadata extracted in the
```
?

```suggestion
A DataLad dataset is created based on the metadata extracted in the
```

adswa replied:
Re-reading this paragraph many times, I feel like I'm not 100% sure what it is telling me. Maybe one introductory sentence in addition helps. Is the gist something like this?

```suggestion
In order to deposit a DataLad dataset next to the original tarball in the remote data store, the following command creates a DataLad dataset  based on the metadata extracted in the
```

adswa replied:
I think the command also misses the ``--id`` parameter and placeholders? I'm getting this when running it:

```
(icf) adina@muninn in /tmp
❱ singularity run -B $STORE_DIR icf.sif deposit_visit_dataset \
  --store-dir $STORE_DIR --store-url https://data.inm-icf.de
usage: deposit_visit_dataset [-h] --id STUDY-ID VISIT-ID [-o PATH] [--store-url URL]
deposit_visit_dataset: error: the following arguments are required: --id
```

adswa replied:
sorry for the flood of comments, I'm realizing more and more things as I'm walking through - I was expecting this to generate a dataset based on the heading, but it doesn't create a standard dataset on my system - just the lightweight representation. Maybe we can reflect this in the heading and description, eg with by placing "dataset" in air quotes or calling it lightweight dataset representation already at the start?

mslw replied:
No need to apologize; thanks a lot for these comments. I agree with the points you make and will make changes accordingly (without using the suggestions directly).

adswa commented on docs/source/user/datalad-generate.rst:
```diff
@@ -0,0 +1,131 @@
+
+.. code-block:: bash
+
+   singularity run -B $STORE_DIR icf.sif deposit_visit_dataset \
+     --store-dir $STORE_DIR --store-url <ICF STORE URL>
+
```
```suggestion
   singularity run -B $STORE_DIR icf.sif deposit_visit_dataset \
     --id <Study ID> <Visit ID> dl-Z03 P000624 --store-dir $STORE_DIR --store-url <ICF STORE URL>

```

mslw replied:
Will do that, but without mixing placeholders and values :wink:

adswa commented on docs/source/user/datalad-generate.rst:
```diff
@@ -0,0 +1,131 @@
+
+   singularity run -B $STORE_DIR icf.sif catalogify_studyvisit_from_meta \
+     --store-dir $STORE_DIR --id <project-ID> <visit ID>
+
```
I think it would be nice to mention that this catalog needs to be subsequently served, or at least point to the README for further instructions - I naively expected the index.html page to display something and initially thought something was wrong.

adswa commented on docs/source/user/datalad-access.rst:
```diff
@@ -0,0 +1,65 @@
+.. code-block::
+
+    'datalad-annex::?type=external&externaltype=uncurl&encryption=none&url=file:///data/group/groupname/local_dicom_store/my-study/P000123_{{annex_key}}'
+
```
I think it would be nice to have an actual fully clone example given here:

```suggestion

A full ``datalad clone`` command could then look like this:

.. code-block::
    datalad clone 'datalad-annex::?type=external&externaltype=uncurl&encryption=none&url=file:///tmp/local_dicom_store/dl-Z03/P000624_{{annex_key}}'  my_clone
    

```

adswa replied:
It may also be worth a note that this command essentially never fails. If I mistype the URL, cloning succeeds, but it tells me 

```
[WARNING] You appear to have cloned an empty repository.                                                          
[WARNING] Cloned /tmp/my_clone but could not find a branch with commits 
```


Which makes it sound like its the dataset's issue, when it just stemmed from a non-existent URL

mslw replied:
Good point, clone from datalad-annex urls does that (related: https://github.com/datalad/datalad-next/issues/373). I'll add a note.

adswa reviewed (approved):
Sorry for missing this - I was neck-deep in too many projects when this PR was posted. I gave it a try as you suggested, and was able to build and clone a repo using my ICF credentials. I left comments along the way as a review.

In general though, this walk-through works, and I feel its a really great addition to have this info. Thanks much for taking the time to write it up!

mslw commented:
Thanks a lot for a thorough review, I will go through the docs again myself and apply the suggestions into account.

mslw commented:
I addressed the suggestions, I'll leave it open for a while and then mark conversations as resolved. Given previous approval, I understand I can merge then.
